engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala

```diff
@@ -54,7 +54,9 @@ class ImportResolver(compiler: Compiler) {
             val importedModules = ir.imports.flatMap {
               case imp: IR.Module.Scope.Import.Module =>
                 val impName = imp.name.name
-                val exp     = ir.exports.find(_.name.name == impName)
+                val exp = ir.exports
+                  .collect { case ex: IR.Module.Scope.Export.Module => ex }
```

Comment:
Again I don't understand why we make the `IR.Module.Scope.Export` a variant type if it has only a single variant `IR.Module.Scope.Export.Module`? This makes us need constructs like this `collect`, but if there is only one variant it seems rather pointless.

Is this because in the future we may get more variants? What would that be?

Reply:
`IR.Error.ImportExport` is the new variant of `Export` added in this PR. And already here the `collect` is useful to only work with the correct exports.

Reply:
Ah ok now I understand. An Error is also a kind of an ImportExport. Ok this is what I missed, now it makes sense :) Thanks for the clarification.